src/headerButton.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import './style/header.css';
+
+class HeaderButton extends Component {
+
+  componentDidMount() {
+    var thisElement = document.getElementById(this.props.id);
+    var anchoredToElement = document.getElementById(this.props.anchoredTo);
+
+    thisElement.onclick = function () { 
+      scrollToItem(anchoredToElement);
+    };
+
+    var scrollToItem = function (item) {
+      var diff=(item.offsetTop-window.scrollY)/20;
+      if(!window._lastDiff){
+          window._lastDiff = 0;
+      }
+
+      if (Math.abs(diff)>2) {
+          window.scrollTo(0, (window.scrollY+diff))
+          clearTimeout(window._TO)
+
+          if(diff !== window._lastDiff){
+              window._lastDiff = diff;
+              window._TO=setTimeout(scrollToItem, 15, item);
```

Comment:
Same here, why 15?

Reply:
The 15 is the execution time as it is recursive, and the 20 is the smoothness it goes through.
I've named those variables to make it clearer.